test/test-utils.ts

```diff
@@ -0,0 +1,21 @@
+import { CfnSubscriptionFilter } from "@aws-cdk/aws-logs";
+import * as cdk from "@aws-cdk/core";
+import { SUBSCRIPTION_FILTER_PREFIX } from "../src/forwarder";
+
+export const findDatadogSubscriptionFilters = (baseConstruct: cdk.Construct) => {
+  return baseConstruct.node
+    .findAll()
+    .filter((construct) => construct.node.id.startsWith(SUBSCRIPTION_FILTER_PREFIX))
+    .map((construct) => {
+      const cfnSubscriptionFilters = construct.node.children.filter(
+        (child) => child instanceof CfnSubscriptionFilter,
+      ) as CfnSubscriptionFilter[];
+
+      expect(cfnSubscriptionFilters).toHaveLength(1);
```

Comment:
Is this because you are only ever expecting to get 1 subscription filter per call to `findDatadogSubscriptionFilters`

Reply:
Correct, I am asserting that the baseConstruct only has a single subscription filter.

Because a future test might want to ensure that a given stack as the baseConstruct has 2 subscription filters (for example, with 2 forwarder arns) I think it makes sense to gate this with a typecheck -- any only assert that there is a single subscription filter if the baseConstruct is a Function. I will push this shortly.

Reply:
I pushed the change but actually.. hold off on reviewing. I will make this a bit clearer.